src/pk.c

```diff
@@ -11916,6 +11916,37 @@ int wolfSSL_EC_KEY_LoadDer_ex(WOLFSSL_EC_KEY* key, const unsigned char* derBuf,
         else {
             ret = wc_EccPublicKeyDecode(derBuf, &idx, (ecc_key*)key->internal,
                 derSz);
+            if (ret < 0) {
+                ecc_key *tmp = (ecc_key*)XMALLOC(sizeof(ecc_key),
+                             ((ecc_key*)key->internal)->heap, DYNAMIC_TYPE_ECC);
+                if (tmp == NULL) {
+                    ret = -1;
+                }
+                else {
+                    /* We now try again as x.963 [point type][x][opt y]. */
+                    ret = wc_ecc_init_ex(tmp, ((ecc_key*)key->internal)->heap,
+                                         INVALID_DEVID);
+                    if (ret == 0) {
+                        ret = wc_ecc_import_x963(derBuf, derSz, tmp);
+                        idx += derSz;
+                        if (ret < 0) {
```

Comment:
This can still leak `tmp` if the init fails. You need to move the free back a brace section.

Please consider re-writing this this section like:

```
ret = wc_EccPublicKeyDecode(derBuf, &idx, (ecc_key*)key->internal, derSz);
if (ret < 0) {
      ecc_key *tmp = (ecc_key*)XMALLOC(sizeof(ecc_key),
                  ((ecc_key*)key->internal)->heap, DYNAMIC_TYPE_ECC);
      if (tmp == NULL) {
            ret = -1;
      }
      else {
            /* We now try again as x.963 [point type][x][opt y]. */
            ret = wc_ecc_init_ex(tmp, ((ecc_key*)key->internal)->heap,
                                 INVALID_DEVID);
            if (ret == 0) {
                  ret = wc_ecc_import_x963(derBuf, derSz, tmp);
                  if (ret == 0) {
                        /* take ownership of new key - set tmp to old key */
                        ecc_key *old = key->internal;
                        key->internal = tmp;
                        tmp = old;

                        idx = derSz;
                  }
                  wc_ecc_free(tmp);
            }
            XFREE(tmp, ((ecc_key*)key->internal)->heap, DYNAMIC_TYPE_ECC);
      }
}
```

Reply:
Yeah, that is so much cleaner. I guess mine was the result of iterative changes based on fixing small things.  Needed a refactor. Thanks!! Will change.